src/pages/index.js

```diff
@@ -100,15 +86,356 @@ const cardSection = new Section(
   ".cards__list"
 );
 
-cardSection.renderItems();
-
-function createCard(cardData) {
-  return new Card(cardData, "#card-template", handleImageClick).generateCard();
+function loadInitialCards() {
+  api.getInitialCards()
+    .then((cards) => {
+      console.log("Fetched cards:", cards);
+      cardSection.renderItems(cards);
+    })
+    .catch((error) => {
+      console.error("Error loading initial cards:", error);
+    });
 }
+  loadInitialCards();
+  
+  function createCard(cardData) {
+    console.log("Creating card with data:", cardData);
+    return new Card(
+      {
+        name: cardData.name,
+        link: cardData.link,
+        id: cardData._id,
+        isLiked: cardData.isLiked
+      },
+      "#card-template",
+      handleImageClick,
+      handleDeleteCard,
+      handleLikeIcon
+    ).generateCard();
+  }
+  /* -------------------------------------------------------------------------- */
+  /*                               Popup Instances                              */
+  /* -------------------------------------------------------------------------- */
+  
+  const imagePopup = new PopupWithImage("#preview-card-modal");
+  imagePopup.setEventListeners();
+  
+  const profileFormPopup = new PopupWithForm(
+    "#edit-profile-modal",
+    handleProfileFormSubmit
+  );
+  profileFormPopup.setEventListeners();
+  
+  const addCardFormPopup = new PopupWithForm(
+    "#add-card-modal",
+    handleAddCardFormSubmit
+  );
+  addCardFormPopup.setEventListeners();
+  
+  const deleteConfirm = new PopupWithConfirm("#delete-card-modal");
+  deleteConfirm.setEventListeners();
+  
+  const avatarPopup = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+  avatarPopup.setEventListeners();
+  
+/* -------------------------------------------------------------------------- */
+/*                               Event Handlers                               */
+/* -------------------------------------------------------------------------- */
 
 function handleImageClick(name, link) {
-    imagePopup.open(name, link);
+  imagePopup.open(name, link);
 }
 
+function handleProfileFormSubmit(formData) {
+  const profileFormSubmitButton = document.querySelector("#profile-edit-submit-button");
+    updateButtonState(profileFormSubmitButton, true);
+
+  api
+    .updateUserInfo({
+      name: formData.name,
+      job: formData.description,
+    })
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        job: userData.about,
+      });
+    })
+    .catch((err) => {
+      console.error("Error updating profile", err);
+    })
+    .finally(() => {
+      setTimeout(() => {
+      updateButtonState(profileFormSubmitButton, false);
+      
+        profileFormPopup.close();
```

Comment:
` profileFormPopup.close();`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) **Please, correct it everywhere in the code**